tests: net: convert to ztest

nniranjhana opened the pull request:
Replaced test points with ztest APIs wherever possible for files in **tests/net/**

JIRA: ZEP-2249

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

jukkar reviewed (changes requested):
Just keep everything in main.c, I do not see much point separating the stuff to two different files.

nniranjhana commented:
@jukkar okay, fixed the commit with the suggested changes.

nashif commented on tests/net/icmpv6/src/main.c:
```diff
@@ -72,51 +75,41 @@ static bool run_tests(void)
+	zassert_true(!(handler_called != 2), "Callbacks not called properly");
 
-	return true;
+	printk("ICMPv6 tests passed\n");
```
remove this, the pass/fail comes from ztest.

nniranjhana replied:
@nashif okay, done

nashif commented on tests/net/ip-addr/src/main.c:
```diff
@@ -223,137 +226,101 @@ static bool run_tests(void)
 
 	printk("IP address print tests passed\n");
```
remove this

nniranjhana replied:
done

nashif commented on tests/net/ip-addr/src/main.c:
```diff
@@ -418,28 +385,18 @@ static bool run_tests(void)
+	zassert_true(net_ipv4_addr_mask_cmp(iface, &match_addr),
+		"IPv4 match failed");
 
 	printk("IP address checks passed\n");
```
remove this

nniranjhana replied:
done

Commit:
tests: icmpv6: convert to ztest

- replaced test points with ztest APIs wherever possible
- replaced CONFIG_MAIN_STACK_SIZE with CONFIG_ZTEST_STACKSIZE
  in config file, reduced size by half

JIRA: ZEP-2249

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

Commit:
tests: ip-addr: convert to ztest

- replaced test points with ztest APIs wherever possible
- replaced CONFIG_MAIN_STACK_SIZE with CONFIG_ZTEST_STACKSIZE
  in config file, reduced the size by half

JIRA: ZEP-2249

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

Commit:
tests: arp: convert to ztest

- replaced test points with ztest APIs wherever possible
- replaced CONFIG_MAIN_STACK_SIZE with CONFIG_ZTEST_STACKSIZE
  in config file

JIRA: ZEP-2249

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

Commit:
tests: mgmt: convert to ztest

- replaced test points with ztest APIs wherever possible
- replaced CONFIG_MAIN_STACK_SIZE with CONFIG_ZTEST_STACKSIZE
  in config file, reduced size to 896

JIRA: ZEP-2249

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

Commit:
tests: mqtt_packet: convert to ztest

- replaced test points with ztest APIs wherever possible

JIRA: ZEP-2249

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

Commit:
tests: http_header_fields: convert to ztest

- replaced test points with ztest APIs wherever possible
- added CONFIG_ZTEST_STACKSIZE=320

JIRA: ZEP-2249

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

Commit:
tests: udp: convert to ztest

- replaced test points with ztest APIs wherever possible
- replaced CONFIG_MAIN_STACK_SIZE with CONFIG_ZTEST_STACKSIZE
  in config file

JIRA: ZEP-2249

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

Commit:
tests: utils: convert to ztest

- replaced test points with ztest APIs wherever possible
- replaced CONFIG_MAIN_STACK_SIZE with CONFIG_ZTEST_STACKSIZE
  in config file, reduced size to 1024

JIRA: ZEP-2249

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

nniranjhana commented:
@jukkar request to review the last commit
@nashif the commits for this PR are done

jukkar reviewed (approved):
LGTM
